sepolicy/mcprepare.te

```diff
@@ -2,6 +2,33 @@
 type mcprepare, domain, coredomain;
 type mcprepare_exec, exec_type, file_type;
 
+# We need to add mcprepare to the dac_overrise_allowed, but we must do so after
+# type mcprepare is declared.
+
+define(`dac_override_allowed', `{
```

Comment:
Looks like it is the same as https://cs.android.com/android/platform/superproject/+/master:system/sepolicy/private/domain.te;l=342-364?q=dac_override_allowed. Will `dac_override_allowed` defined here override `dac_override_allowed` at system/sepolicy? Is it permitted to use different name than `dac_override_allowed`, for example `dac_override_allowed_maru`? If we cant' use other name, do we need update it when upgrading Android base version?

Reply:
> Will dac_override_allowed defined here override dac_override_allowed at system/sepolicy?

It will not override it unfortunately. dac_override_allowed is first initialized in system/sepolicy without mcprepare and then all of the normal checks are run (including the failing one). After that dac_override_allowed is re-initialized with mcprepare added and line 30 validates the neverallow rule against dac_override with mcprepare added as an exception.

- If a device is already passing this check in system/policy, then it will also pass this check.

- If a device is failing the check in system/sepolicy, then the maintainer can go and remove the check from system/sepolicy and the rule is still enforced. We could even add a sed command automatically removing the neverallow check to our build process or run post repo sync.

>  Is it permitted to use different name than dac_override_allowed, for example dac_override_allowed_maru?

We can use a different name. This would also probably make this more readable. We would just modify the neverallow rule to use the new variable. I can update the code with this change.

> If we cant' use other name, do we need update it when upgrading Android base version?

Redgardless of whether we use a different name or not, base Android upgrades could possibly cause us to upgrade this. If dac_override_allowed has new types added to it, then dac_override_allowed_maru would need to also need those types added to it. 
I could go in and add comments explaining this to the code in the form of a TODO so we know in future up-porting to check this variable for changes.



Anyways, if these changes sound good to you, I can add them to the PR.

Reply:
I didnt see your comment over at https://github.com/maruos/maruos/issues/154#issuecomment-961193695 before posting the last reply. I see your point. That is giving more power than I realized. Let me do some additional reading on this.

Here is the avc:
libsepol.report_failure: neverallow on line 1385 of system/sepolicy/public/domain.te (or line 11832 of policy.conf) violated by allow mcprepare mcprepare:capability { dac_override };
libsepol.check_assertions: 1 neverallow failures occurred
Error while expanding policy
/var/maru/devel/maru-0.7-upstream/out/host/linux-x86/bin/checkpolicy:  loading policy configuration from /var/maru/devel/maru-0.7-upstream/out/target/product/taimen/obj/ETC/sepolicy_neverallows_intermediates/policy.conf

Reply:
I just ran a test where I removed dac_override from mcprepare and reran the build without this change. This got me a clean build and past the dac_override check. I booted it up, launched the container, and connected via wireless display. And everything seems to be working. This seems like the way to go. Remove mcprepare from using dac_override in the first place. I can update this PR with this fix if you want to give it a try.